AWSSDKSwiftCLI/Tests/AWSSDKSwiftCLITests/Models/ReleaseNotesBuilderTests.swift

```diff
@@ -0,0 +1,218 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+@testable import AWSSDKSwiftCLI
+import AWSCLIUtils
+import XCTest
+
+/*
+ *  Regression tests for protection against change in generated release notes markdown content.
+ */
+class ReleaseNotesBuilderTests: XCTestCase {
+    /* Reusable feature strings */
+
+    // New feature 1
+    private let feature1 = """
+    {
+        "releaseNotes": "New feature description A.",
+        "featureMetadata": {
+            "trebuchet": {
+                "featureId": "feature-id-a",
+                "featureType": "NEW_FEATURE",
+            }
+        }
+    }
+    """
+
+    // New feature 2
+    private let feature2 = """
+    {
+        "releaseNotes": "New feature description B.",
+        "featureMetadata": {
+            "trebuchet": {
+                "featureId": "feature-id-b",
+                "featureType": "NEW_FEATURE",
+            }
+        }
+    }
+    """
+
+    // Documentation update
+    private let feature3 = """
+    {
+        "releaseNotes": "Doc update description C.",
+        "featureMetadata": {
+            "trebuchet": {
+                "featureId": "feature-id-c",
+                "featureType": "DOC_UPDATE",
+            }
+        }
+    }
+    """
+
+    // Feature with null releaseNotes field
+    private let feature4 = """
+    {
+        "releaseNotes": null,
```

Comment:
This is how preview builds encode release notes: the `releaseNotes` key is present with a `null` value?

Do we need to test where the `releaseNotes` key is omitted entirely?

Reply:
(I _think_ an autogenerated Swift decoder will decode either case to a `String?` value of `nil`.)

Reply:
Yeah, it's `releaseNotes` key present but with a `null` value.

Reply:
And yeah I think in either case Swift decoder just assigns `nil` as well